pandas/core/frame.py

```diff
@@ -3043,6 +3043,11 @@ def __getitem__(self, key):
 
         # Do we have a (boolean) DataFrame?
         if isinstance(key, DataFrame):
+            if not (key.index.equals(self.index) and key.columns.equals(self.columns)):
```

Comment:
I am not sure is equals is the right thing to do here.

```
idx = Index([1, 2, 3])

idx2 = Index([3, 2, 1])

idx.equals(idx2)
```

This returns False but we can certainly align bot indexes

Reply:
I agree that they are alignable. I thought the idea was to not accept alignables anymore. I just wanted to make an exception were alignables are accepted if they are already aligned, so that stuff like `df[df > 0]` is still possible.